internal/auth0/resourceserver/resource_scope.go

```diff
@@ -49,45 +49,60 @@ func NewScopeResource() *schema.Resource {
 
 func createResourceServerScope(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
 
-	resourceServerID := data.Get("resource_server_identifier").(string)
+	resourceServerIdentifier := data.Get("resource_server_identifier").(string)
 	scope := data.Get("scope").(string)
 	description := data.Get("description").(string)
 
-	currentScopes, err := api.ResourceServer.Read(resourceServerID)
+	mutex.Lock(resourceServerIdentifier)
+	defer mutex.Unlock(resourceServerIdentifier)
+
+	existingAPI, err := api.ResourceServer.Read(resourceServerIdentifier)
 	if err != nil {
 		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
 			data.SetId("")
 			return nil
 		}
+
 		return diag.FromErr(err)
 	}
 
-	scopes := append(currentScopes.GetScopes(), management.ResourceServerScope{
+	data.SetId(fmt.Sprintf(`%s::%s`, resourceServerIdentifier, scope))
+
+	for _, apiScope := range existingAPI.GetScopes() {
+		if apiScope.GetValue() == scope {
+			return readResourceServerScope(ctx, data, meta)
+		}
+	}
```

Comment:
This is the core of the fix, the other changes are mostly to increase readibility.